autoscheduler/scraper/management/commands/scrape_courses.py

```diff
@@ -1,6 +1,105 @@
+import asyncio
+import time
+from typing import List
 from django.core.management import base
+from scraper.banner_requests import BannerRequests
+from scraper.models import Course, Instructor, Section, Meeting
+from scraper.models.course import generate_course_id
+
+def parse_section(course,i):
+    #gets values to create object
+    section_id = int(course['id'])
+    section_subject = course['subject']
+    course_number = course['courseNumber']
+    section_number = int(course['sequenceNumber'])
+    section_term_code = course['term']
+    section_min_credits = course['creditHourLow']
+    section_max_credits = course['creditHourHigh']
+    section_max_enrollment = course['maximumEnrollment']
+    section_current_enrollment = course['enrollment']
+    section_instructor = i
+    #creates and saves section object
+    s = Section(id=section_id, subject=section_subject, course_num=course_number, section_num=section_number, term_code=section_term_code, min_credits=section_min_credits, max_credits=section_max_credits, max_enrollment=section_max_enrollment, current_enrollment=section_current_enrollment, instructor=section_instructor)
+    s.save()
+    #calls parse meeting for each meeting time in the section
+    for i in course['meetingsFaculty']:
+        parse_meeting(i)
+
+def parse_meeting(i):
+    #gets values to create object
+    meeting_id = ''
+    meeting_crn = i['meetingTime']['courseReferenceNumber']
+    meeting_building = i['meetingTime']['building']
+    #checks which days classes are on and adds them to attend_days
+    meeting_class_days = ['monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday']
+    attend_days = []
+    for j in meeting_class_days:
+        if i['meetingTime'][j] == True:
+            attend_days.append(j)
+    meeting_start_time = i['meetingTime']['beginTime']
+    meeting_end_time = i['meetingTime']['endTime']
+    meeting_class_type = i['meetingTime']['meetingType']
+    meeting_section = ''#this should be an object
+    #creates and saves meeting object
+    m = Meeting(id=meeting_id, crn=meeting_crn, building=meeting_building, meeting_days=attend_days, start_time=meeting_start_time, end_time=meeting_end_time, meeting_type=meeting_class_type, section=meeting_section)
+    m.save()
+
+def parse_instructor(course):
+    #gets values to create object
+    for i in course['faculty']:
+        instructor_id = i['bannerId']
```

Comment:
Since the bannerId isn't actually unique to a professor, and it looks like the `id` object is always null in the section data from Banner, I think we should change it to use the instructor's name as the primary key. We can't do the email, since some professors don't have an email listed (not sure how that makes any sense but w/e lol)

Reply:
Ok. Got it